cmake/extensions.cmake

```diff
@@ -917,6 +917,12 @@ function(zephyr_library_link_libraries_ifdef feature_toggle item)
   endif()
 endfunction()
 
+function(zephyr_syscall_include_ifdef feature_toggle item)
```

Comment:
Can we have a zephyr_syscall_include() function as well which isn't conditional on some kconfig?

Reply:
i thought of this but i was a bit skeptical. 
my reason was: If we don't know when excatly we need these extra syscalls, we shouldn't be doing this in the first place.
I felt its not correct to include these syscalls by default and hence i didn't create a  zephyr_syscall_include().
Do you see a need for this function?

Reply:
yes, if the application defines its own syscalls and needs to use them unconditionally. this would be in the application build files.